useragent/useragent.go

```diff
@@ -82,23 +93,22 @@ func UserAgent(binaryNameCapitalized string, version, commit, buildTime string,
 	return builder.String()
 }
 
-func UserAgentWithBeatTelemetry(binaryNameCapitalized string, version string, mode AgentManagementMode, unprivileged AgentUnprivilegedMode) (string, error) {
+func UserAgentWithBeatTelemetry(binaryNameCapitalized string, version string, mode AgentManagementMode, unprivileged AgentUnprivilegedMode) string {
 	var builder strings.Builder
 	builder.WriteString("Elastic-" + binaryNameCapitalized + "/" + version + " ")
 	uaValues := []string{
 		runtime.GOOS,
 		runtime.GOARCH,
-		mode.String(),
-		unprivileged.String(),
+	}
+	if mode != AgentManagementModeUnknown {
+		uaValues = append(uaValues, mode.String())
+	}
+	if unprivileged != AgentUnprivilegedModeUnknown {
+		uaValues = append(uaValues, unprivileged.String())
 	}
 	builder.WriteByte('(')
 	builder.WriteString(strings.Join(uaValues, "; "))
 	builder.WriteByte(')')
 
-	// Ensure the user agent string does not exceed 100 characters
-	userAgent := builder.String()
-	if len(userAgent) > 100 {
```

Comment:
Having this check was a requirement from https://github.com/elastic/ingest-dev/issues/5333 considering our proxy logs are limiting the user agent size to 100.

Reply:
I'd be ok to add a failing test but it shouldn't be in the API - what's the consumer supposed to do it if it fails?  The previous implementation still returned the calculated user agent but using that when error is not nil is too non-idiomatic.

Reply:
added this as a test